### PR TITLE
Feature/mnist conversion

### PR DESCRIPTION
**Pull request by leoTiez**
Solves #3

### CODE REVIEW

File: download-data.sh [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
-unameOut="$(uname -s)"
+OS_NAME="$(uname -s)"
 
-case "${unameOut}" in
-    Linux*)     sudo apt-get update && sudo apt-get install curl;;
-    CYGWIN*)    sudo apt-get update && sudo apt-get install curl;;
-    MINGW*)     sudo apt-get update && sudo apt-get install curl;;
-esac
+if [[ "$OS_NAME" == "Linux" ]]; then
+    sudo apt-get update && sudo apt-get install curl
```
Comment (adrianc-a) [inline download-data.sh:6]: hmm. ok, but i feel like it's easier to just assume curl is installed

File: download-data.sh [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
+fi
 
-rm -rf data/
-mkdir data
-curl https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz -o data/cifar-10-batches-py.tar.gz
-tar -xvzf data/cifar-10-batches-py.tar.gz -C data/
-rm -rf data/cifar-10-batches-py.tar.gz
+DATA_DIRECTORY="data"
+if [[ ! -d "$DATA_DIRECTORY" ]]; then
+  mkdir data
```
Comment (adrianc-a) [inline download-data.sh]: let's use the variables we create.

File: download-data.sh [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
+CIFAR_100_DIRECTORY="data/cifar-100-python"
+if [[ ! -d "$CIFAR_100_DIRECTORY" ]]; then
+    curl https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz -o data/cifar-100-batches-py.tar.gz
+    tar -xvf data/cifar-100-batches-py.tar.gz -C data/
+    rm -rf data/cifar-100-batches-py.tar.gz
+fi
+
+MNIST_DIRECTORY="data/mnist"
+if [[ ! -d "$MNIST_DIRECTORY" ]]; then
+    mkdir data/mnist
```
Comment (adrianc-a) [inline download-data.sh]: here too

File: load_data.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    :return: The cifar data as a dictionary
+    """
+    with open(str(file_path + file_name), 'rb') as fo:
+        data_dict = cPickle.load(fo)
+    data_dict['data'] = np.asarray(data_dict['data'], dtype=dtype)
+    data_dict['labels'] = np.asarray(data_dict['labels'], dtype=dtype)
+    return data_dict
+
+
+def pre_process_cifar_data(cifar_data, dtype):
```
Comment (adrianc-a) [inline load_data.py]: preprocess.

**Review by adrianc-a — APPROVED** [review]
so i fixed most of the small things i pointed out. 

i can run on my machine and the DL script works, and I can get CRNN.py running with the data.

File: download-data.sh (hunk 1 above)
Comment (leoTiez) [reply download-data.sh:6]: It's not per default on Linux, that's why I added it.

File: download-data.sh (hunk 2 above)
Comment (leoTiez) [reply download-data.sh]: Very good point! :+1:

**Pull request merged**